src/Parallel/Printf.hpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines Parallel::printf for writing to stdout
+
+#pragma once
+
+#include <charm++.h>
+#include <sstream>
+#include <string>
+#include <type_traits>
+
+namespace Parallel {
+namespace detail {
+// Print pointers
+template <typename T,
+          typename std::enable_if_t<
+              std::is_fundamental<std::decay_t<
+                  std::remove_pointer_t<std::decay_t<T>>>>::value or
+              std::is_pointer<T>::value> * = nullptr>
+inline constexpr T convert_to_printable_type(T &&t) {
+  return t;
+}
+
+// Print objects with stream operator
+template <typename T,
+          typename std::enable_if_t<
+              not std::is_fundamental<std::decay_t<
+                  std::remove_pointer_t<std::decay_t<T>>>>::value and
+              not std::is_pointer<T>::value> * = nullptr>
+inline std::string convert_to_printable_type(T &&t) {
+  std::stringstream ss;
+  ss << t;
+  return ss.str();
+}
+
+// Print references to pointers and fundamentals
+template <typename T,
+          typename std::enable_if_t<
+              std::is_fundamental<std::decay_t<
+                  std::remove_pointer_t<std::decay_t<T>>>>::value or
+              std::is_pointer<std::decay_t<T>>::value> * = nullptr>
+inline constexpr T get_printable_type(T &&t) {
+  return t;
+}
+
+inline const typename std::string::value_type *get_printable_type(
```

Comment:
This is something we can try to control with ClangFormat. Right now it moves all to one, whichever is more prominent in the file. Do you want me to see if there is an option and if it works in CF?